free-1.md

```diff
@@ -1,68 +1,170 @@
+
+
+
 ### Operation (semantics)
+
 _What might happen when a program runs? How does a program interact with the
+
 user? What kinds of errors might occur, and how might they be communicated to
+
 the user?_
 
 
+
+When the program runs, it would generate any created tiles, then repeat the patterns and use the map specifications to create each piece, then stitch them together and output an image.
+
+If any area is left blank, it should probably color it in with an obnoxiously bright color so the user knows to cover that area.
+
+
+
 ### Expressiveness
+
 _What should be easy to do in this language? What should be possible, but
+
 difficult? What should be impossible or very difficult?_
 
 
+
+This all depends on constraints. Resizing things may prove difficult.
+
+
+
 ### Related work
+
 _Are there any other DSLs in this domain? If not, describe how you know there
+
 aren't and conjecture why not. If so, describe them and provide links. How well 
+
 do they address the need? Are there any particularly admirable qualities of the
+
 language? Are there parts of the language you think could be improved?_
 
 
+
+There are loads of RPG mapmakers. Most of them have pre-defined tile sizes, but with enough searching it may be possible to find one that has this behavior; however, overall they give game developers enough tools to succeed. 
+
+
+
 ## The Project
+
 This section examines whether the idea makes for a good CS 111 project.
 
 
+
+This is probably too big an idea to actually pull off, but parts of it might get done. I'm also not sure if it's enough of a simulation language to be considered a good DSL.
+
+
+
 ### Suitability
+
 _If someone were to work on this project, what percentage of their time would be
+
 spent directly engaging in the **language** aspects of this project (e.g.,
+
 making language design decisions), as opposed to "systems" aspects of the
+
 project (e.g., implementing a complicated semantics that doesn't require a lot
+
 of language design)?_
 
 
+
+Well, the "words" would be the map areas and the tiles. How these areas are populated would be a design decision; perhaps if a tile is bigger than an area, the user should be able to specify where the cutoff is; likewise, maybe they should be allowed to "move around" the tiling once the region is populated. In that case, would it still be a DSL? Is this enough of a language design?
+
+
+
 ### Scope
+
 _How big an idea is this? How ambitious is this project?_
 
 
+
+This is fairly ambitious. Having tiles input rather than allowing them to be made would probably help to scale it down to an attainable region.
+
+
+
 ### Benefits and drawbacks
+
 _Why might this be a good idea for a project? Why might this not be a good idea 
+
 project?_
 
+
+
+This is sort of like a simulation and sort of like a creativity language. It's also useful and might be fun to use. But, it might not be enough of a DSL, and there may be something out there like this already.
+
```

Comment:
This is a good and interesting idea. The part about irregular shapes / tiles makes it especially interesting. If you were to go forward with this idea, I would ask you to first focus on the language aspects, e.g., how to encode / specify the map. This might mean focusing on a text-based language or an intermediate DSL first. Then, if there's time, you could work on integrating this language with a UI. If there's not time this semester, you could always make a quick mock up that would still convey the idea.